dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -787,25 +814,36 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
         Optional<SourceFile> mainDescriptor = filteredSourcefiles.stream().filter((sourceFile -> Objects.equals(sourceFile.getPath(), primaryDescriptorFilePath))).findFirst();
 
         boolean isValid = true;
+        boolean safe = false;
         StringBuilder validationMessage = new StringBuilder();
         Map<String, String> validationMessageObject = new HashMap<>();
 
         if (mainDescriptor.isPresent()) {
-            Yaml yaml = new Yaml();
-            String content = mainDescriptor.get().getContent();
-            if (content == null || content.isEmpty()) {
-                isValid = false;
-                validationMessage.append("Primary descriptor is empty.");
-            } else if (!content.contains("class: Workflow")) {
+            try {
+                Yaml safeYaml = new Yaml(new SafeConstructor());
+                // This should throw an exception if there are unexpected blocks
+                safeYaml.load(mainDescriptor.get().getContent());
+                safe = true;
+            } catch (Exception e) {
                 isValid = false;
-                validationMessage.append("A CWL workflow requires 'class: Workflow'.");
-                if (content.contains("class: CommandLineTool") || content.contains("class: ExpressionTool")) {
-                    String cwlClass = content.contains("class: CommandLineTool") ? "CommandLineTool" : "ExpressionTool";
-                    validationMessage.append(" This file contains 'class: ").append(cwlClass).append("'. Did you mean to register a tool?");
+            }
+            if (safe) {
```

Comment:
git doesn't display this terribly well, but I just did the safety test and use the results to bypass parsing.